[jit] Add unsafeRemoveConstant for ClassType

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29220 [quant][graphmode] Enable foldbn tests
* #29219 [jit] Expose class constant through `attr` and `setattr` in object
* **#30787 [jit] Add unsafeRemoveConstant for ClassType**
* #31012 [jit] Fix getConstant
* #29218 [jit] Preserve constant from ConcreteModuleType to ClassType

Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

jerryzh168 committed:
[jit] Add unsafeRemoveConstant for ClassType

Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 785bb22:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=785bb22aadb2a919ef434cf0a0ba642320cc338c).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3919560) binary_linux_libtorch_3_7m_cpu_devtoolset7_shared-with-deps_build (1/1)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72954283))
```
Dec 12 05:57:03        ^~~~~~~~~~~~~~~~~~~~~
Dec 12 05:57:03 /pytorch/torch/csrc/jit/script/class_type.cpp:143:3: note: in expansion of macro ‘TORCH_INTERNAL_ASSERT’
Dec 12 05:57:03    TORCH_INTERNAL_ASSERT(constantNames_.size() == constantValues_.size());
Dec 12 05:57:03    ^~~~~~~~~~~~~~~~~~~~~
Dec 12 05:57:03 /pytorch/torch/csrc/jit/script/class_type.cpp:145:24: error: ‘constantNames_’ was not declared in this scope
Dec 12 05:57:03    for (const auto& c : constantNames_) {
Dec 12 05:57:03                         ^~~~~~~~~~~~~~
Dec 12 05:57:03 /pytorch/torch/csrc/jit/script/class_type.cpp:152:14: error: ‘constantNames_’ was not declared in this scope
Dec 12 05:57:03    if (pos >= constantNames_.size()) {
Dec 12 05:57:03               ^~~~~~~~~~~~~~
Dec 12 05:57:03 /pytorch/torch/csrc/jit/script/class_type.cpp:155:10: error: ‘constantValues_’ was not declared in this scope
Dec 12 05:57:03    return constantValues_[pos];
Dec 12 05:57:03           ^~~~~~~~~~~~~~~
Dec 12 05:57:03 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/builtin_functions.cpp.o
Dec 12 05:57:04 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/canonicalize_modified_loop.cpp.o
Dec 12 05:57:04 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/edit_distance.cpp.o
Dec 12 05:57:05 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/logging.cpp.o
Dec 12 05:57:05 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/module.cpp.o
Dec 12 05:57:05 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/object.cpp.o
Dec 12 05:57:05 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/jit_exception.cpp.o
Dec 12 05:57:06 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/source_range_serialization.cpp.o
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Add unsafeRemoveConstant for ClassType"


Summary:
This is needed when we fuse conv bn modules,
where we need to rewrite a constant bias (None) of conv to an attribute
bias of Tensor

Test Plan:
build/bin/test_jit

Reviewers:
suo, zdevito

Subscribers:

Tasks:

Tags:

Differential Revision: [D18846850](https://our.internmc.facebook.com/intern/diff/D18846850)

[ghstack-poisoned]
